src/rgw/rgw_admin.cc

```diff
@@ -4017,7 +4045,50 @@ int main(int argc, char **argv)
       cout << std::endl;
     }
     return 0;
-
+  case OPT_ROLE_CREATE:
+    {
+      if (role_name.empty()) {
+        cerr << "Empty role name" << std::endl;
+        return -EINVAL;
+      }
```

Comment:
@cbodley : path can be empty since it is optional. It is set to "/" if the user doesn't set it to anything (according to AWS create role definition.)